src/entities.rs

```diff
@@ -490,8 +493,8 @@ impl Entities {
         }
     }
 
-    fn needs_flush(&mut self) -> bool {
-        *self.free_cursor.get_mut() != self.pending.len() as isize
+    pub fn needs_flush(&self) -> bool {
```

Comment:
This judgement isn't meaningful if the caller doesn't have unique access.

Reply:
I also serialize the free_cursor now, so this is not necessary anymore.